vm/vm.go

```diff
@@ -143,6 +143,10 @@ type VM struct {
 	stop  chan struct{}
 }
 
+func (vm *VM) ImmutableState(_ context.Context) (state.Immutable, error) {
```

Comment:
```suggestion
// TODO: update VM ImmutableState to return an isolated trie view instead of
// reading directly from the on-disk view, which may change mid operation.
// ref. https://github.com/ava-labs/avalanchego/pull/3358/files
func (vm *VM) ImmutableState(_ context.Context) (state.Immutable, error) {
```

Could we also move this to `vm/resolutions.go` to follow the current code organization pattern?

Reply:
Not sure what happened here but the suggestion has been applied.